### PR TITLE
Enable i64 bit indexing for unary MLIR kernels

### PR DESCRIPTION
**Pull request by kushanam**
Enabling unary index64 kernels
@CC @frgossen

### CODE REVIEW

**Review by frgossen — COMMENTED** [review]
This is hard to review with all the unrelated formatting changes. Can you fix the formatting?

**Review by frgossen — CHANGES_REQUESTED** [review]
This is hard to review with all the unrelated formatting changes. Can you fix the formatting?

**Comment by kushanam** [comment]
> This is hard to review with all the unrelated formatting changes. Can you fix the formatting?

Oh sorry, this was suggested by VS code bazel plugin formatter. I changed back the formatting.

File: tensorflow/core/kernels/mlir_generated/BUILD [hunk 1]
```diff
@@ -1377,9 +1852,9 @@ gpu_kernel_library(
     ] * 14,
     tile_size = "256",
     types = ["i1"] * 14 + ["i8"] * 14 + ["i16"] * 14 + ["i32"] * 14 +
-            ["i64"] * 14 + ["ui8"] * 14 + ["ui16"] * 14 + ["ui32"] * 14 +
-            ["ui64"] * 14 + ["f16"] * 14 + ["f32"] * 14 + ["f64"] * 14 +
-            ["c64"] * 14 + ["c128"] * 14,
+             ["i64"] * 14 + ["ui8"] * 14 + ["ui16"] * 14 + ["ui32"] * 14 +
```
Comment (frgossen) [inline tensorflow/core/kernels/mlir_generated/BUILD:1855]: nit: no change, please fix formatting.

File: tensorflow/core/kernels/mlir_generated/BUILD [hunk 2]
```diff
@@ -1408,7 +1883,7 @@ gpu_kernel_library(
         "f64",
     ],
     # TODO(b/178388085): Enable unrolling after vectorization is fixed.
-    # unroll_factors = "4",
+    # unroll_factors  =  "4",
```
Comment (frgossen) [inline tensorflow/core/kernels/mlir_generated/BUILD:1886]: nit: no change, please fix formatting.

File: tensorflow/core/kernels/mlir_generated/BUILD [hunk 3]
```diff
@@ -641,6 +756,22 @@ gpu_kernel_library(
     unroll_factors = "2",
 )
 
+# gpu_kernel_library(
+#     name  =  "gpu_angle_kernels_experimental",
```
Comment (frgossen) [inline tensorflow/core/kernels/mlir_generated/BUILD:760]: Is this intentionally added?

File: tensorflow/core/kernels/mlir_generated/BUILD [hunk 4]
```diff
@@ -601,6 +673,25 @@ gpu_kernel_library(
...
+        "i8",
+        "i16",
+        "f16",
+        "f32",
+        "f64",
+        "i64",
+    ],
+    op = "abs",
+    test_tags = [
+        "noasan",  # TODO(b/248071004): JIT-compiled code will not be instrumented.
```
Comment (frgossen) [inline tensorflow/core/kernels/mlir_generated/BUILD:688]: Is this needed? If so, is it needed everywhere?

**Review by frgossen — CHANGES_REQUESTED** [review]
Looks good. Let's land this after we have the test case running!

File: tensorflow/core/kernels/mlir_generated/BUILD [hunk 5]
```diff
@@ -469,6 +507,28 @@ tf_cuda_cc_test(
     ],
 )
 
+tf_cuda_cc_test(
```
Comment (frgossen) [inline tensorflow/core/kernels/mlir_generated/BUILD:510]: Does this belong here? Is it not already part of https://github.com/tensorflow/tensorflow/pull/58045

**Review by frgossen — CHANGES_REQUESTED** [review]
Nice! Can you clarify in the PR name / description that this adds i64-indexing support for **unary** kernels behind an experimental flag?

File: tensorflow/core/kernels/mlir_generated/BUILD (hunk 5 above)
Comment (kushanam) [reply tensorflow/core/kernels/mlir_generated/BUILD]: Addressed.

File: tensorflow/core/kernels/mlir_generated/BUILD (hunk 4 above)
Comment (kushanam) [reply tensorflow/core/kernels/mlir_generated/BUILD:666]: It was added priorly and only for this kernel.

File: tensorflow/core/kernels/mlir_generated/BUILD (hunk 2 above)
Comment (kushanam) [reply tensorflow/core/kernels/mlir_generated/BUILD]: done.

File: tensorflow/core/kernels/mlir_generated/BUILD (hunk 1 above)
Comment (kushanam) [reply tensorflow/core/kernels/mlir_generated/BUILD]: done.

File: tensorflow/core/kernels/mlir_generated/BUILD (hunk 3 above)
Comment (kushanam) [reply tensorflow/core/kernels/mlir_generated/BUILD]: removed.

**Review by frgossen — COMMENTED** [review]
Thanks for adding these!

**Review by frgossen — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @kushanam Can you please resolve conflicts? Thank you!

**Pull request closed without merging**